Shared/mods/deathmatch/logic/physics/CLuaPhysicsBaseManager.h

```diff
@@ -0,0 +1,65 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        mods/shared_logic/logic/lua/CLuaPhysicsBaseManager.h
+ *  PURPOSE:     Lua physics base manager class
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+template <class T>
+class CLuaPhysicsBaseManager;
+
+#pragma once
+
+template <class T>
+class CLuaPhysicsBaseManager
+{
+public:
+    CLuaPhysicsBaseManager(EIdClassType idClass) : m_IdClass(idClass){};
+    ~CLuaPhysicsBaseManager() {}
+
+    T GetFromScriptID(unsigned int uiScriptID)
+    {
+        T pElement = (T)CIdArray::FindEntry(uiScriptID, m_IdClass);
+        if (!pElement)
+            return nullptr;
+        
+        if (!ListContains(m_elementsList, pElement))
+            return nullptr;
+        return pElement;
+    }
+
+    virtual void Remove(T pElement, bool deleteFromList = true) = 0;
+
+    void RemoveAll(CResource* pResource)
+    {
+        std::vector<T> elementsToRemove;
```

Comment:
What the purpose of using an additional vector? Why don't simple remove elements in the place?